Sources/PT.PM.Common/UstConstantFolder.cs

```diff
@@ -144,8 +143,15 @@ private FoldResult TryFoldBinaryOperatorExpression(BinaryOperatorExpression bina
                     return result;
                 }
             }
-            else if (leftValue is long leftInt && rightValue is long rightInt)
+            else if ((leftValue is long || leftValue is int)
+                && (rightValue is long || rightValue is int))
```

Comment:
You forgot about `BigInt`. Moreover, I think it's better to get rid of type casts here and to split this method into 3 ones: for `int`, `long` and `BigInt` types.